pkg/steps/cluster_claim.go

```diff
@@ -63,7 +63,13 @@ func (s *clusterClaimStep) run(ctx context.Context) error {
 	}
 	clusterClaim, err := acquireCluster(ctx, *s.clusterClaim, s.hiveClient, s.client, *s.jobSpec)
 	if err != nil {
-		return results.ForReason("acquiring_cluster_claim").ForError(err)
+		acquireErr := results.ForReason("acquiring_cluster_claim").ForError(err)
+		// always attempt to delete claim if one exists
+		var releaseErr error
+		if clusterClaim != nil {
```

Comment:
Maybe push line76 (the logging line) to the function `releaseCluster` so that we can see it in both cases of release?